ui_testing/testcases/basic_tests/test05_testplans.py

```diff
@@ -55,4 +55,61 @@ def test001_test_plan_delete_testunit(self):
         
         self.assertFalse(deleted_test_unit_found)
 
+    def test012_test_unit_update_version_in_testplan(self):
+        '''
+        LIMS-3703
+        Create two testplans the first one with the testunit before updating its version, while
+        the second is after updating the version (category and iteration)
+        The updates should be only available in the second testplan
+        '''
+        
+        # get all the testunits and choose a random testunit to create the first testplan with
+        testunits = self.get_all_testunits()
+        random_testunit = random.choice(testunits)
+        self.base_selenium.LOGGER.info('A random testunit is chosen, its name: {}, category: {} and number of iterations: {}'.format(random_testunit['name'], random_testunit['categoryName'], random_testunit['iterations']))
+
+        # get all articles and choose a random one to take its information
+        articles = self.get_all_articles()
+        random_article = random.choice(articles)
+        self.base_selenium.LOGGER.info('A random article is chosen, its name: {} and its material type: {}'.format(random_article['name'], random_article['materialType']))
+
+        # create the first testplan
+        first_testplan_name = self.test_plan.create_new_test_plan(material_type=random_article['materialType'], article=random_article['name'], test_unit=random_testunit['name'])
+        self.base_selenium.LOGGER.info('New testplan is created successfully with name: {}, article name: {} and material type: {}'.format(first_testplan_name, random_article['name'], random_article['materialType']))
+
+        # go to testplan edit to get the number of iterations and testunit category
+        first_testplan_testunit_category, first_testplan_testunit_iteration = self.test_plan.get_testunit_category_and_iterations(first_testplan_name)
+
+        # go to testunits active table and search for this testunit-
+        self.test_unit_page.get_test_units_page()
+        self.base_selenium.LOGGER.info('Navigating to testunit {} edit page'.format(random_testunit['name']))
```

Comment:
1.  Use the API to get a complete test plan. 
2. Use API to get the related test units of this test plan 
3- Edit this test unit 
4- Verify the test plan is okay 
5- Create a new test plan with this test unit
6- Verify this new test plan

Reply:
I did this, but I found a problem in assertion, because testunits can be created with non unique names, so when choosing a testunit from the drop down menu when creating a testplan, it might select a testunit with the same name as the changed one, but not the changed one itself.

Reply:
@SalmaaRagab Nice catch, but I can see that the current implementation is having the same problem. You r choosing a random test unit then your searching for it using the name. 

I suggest using an API to create a test unit and an API to create the first test plan.

Reply:
I think having an API call to create a testunit or testplan would be too much

Reply:
@0xIslamTaha I suggest creating a new testunit for this testcase, then creating a testplan for it, going back to edit this testunit, then creating a new testplan.
I know this could be so much, but I think it's easier than creating an API call to create the new testunit.

Reply:
@SalmaaRagab How come? API should be a piece of cake. No?

Reply:
No, it's not.
In order to create a testunit/testplan the request is large and needs some ids for the material type and the article.
This is how the request should look like:
![image](https://user-images.githubusercontent.com/53828649/71578649-b29fcc00-2b01-11ea-992b-3904dbbdeaa0.png)